integrations/amazon_bedrock/src/haystack_integrations/components/generators/amazon_bedrock/chat/chat_generator.py

```diff
@@ -0,0 +1,243 @@
+        Amazon Bedrock client.
+
+        Note that the AWS credentials are not required if the AWS environment is configured correctly. These are loaded
+        automatically from the environment or the AWS configuration file and do not need to be provided explicitly via
+        the constructor.
+
+        :param model: The model to use for generation. The model must be available in Amazon Bedrock. The model has to
+        be specified in the format outlined in the Amazon Bedrock [documentation](https://docs.aws.amazon.com/bedrock/latest/userguide/model-ids-arns.html).
+        :param aws_access_key_id: AWS access key ID.
+        :param aws_secret_access_key: AWS secret access key.
+        :param aws_session_token: AWS session token.
+        :param aws_region_name: AWS region name.
+        :param aws_profile_name: AWS profile name.
+        :param generation_kwargs: Additional generation keyword arguments passed to the model. The defined keyword
+        parameters are specific to a specific model and can be found in the model's documentation. For example, the
+        Anthropic Claude generation parameters can be found [here](https://docs.anthropic.com/claude/reference/complete_post).
+        :param stop_words: A list of stop words that stop model generation when encountered. They can be provided via
+        this parameter or via models generation_kwargs under a model's specific key for stop words. For example, the
+        Anthropic Claude stop words are provided via the `stop_sequences` key.
+        :param streaming_callback: A callback function that is called when a new chunk is received from the stream.
+        By default, the model is not set up for streaming. To enable streaming simply set this parameter to a callback
+        function that will handle the streaming chunks. The callback function will receive a StreamingChunk object and
+        switch the streaming mode on.
+        """
+        if not model:
+            msg = "'model' cannot be None or empty string"
+            raise ValueError(msg)
+        self.model = model
+        try:
+            session = self.get_aws_session(
+                aws_access_key_id=aws_access_key_id,
+                aws_secret_access_key=aws_secret_access_key,
+                aws_session_token=aws_session_token,
+                aws_region_name=aws_region_name,
+                aws_profile_name=aws_profile_name,
+            )
+            self.client = session.client("bedrock-runtime")
+        except Exception as exception:
+            msg = (
+                "Could not connect to Amazon Bedrock. Make sure the AWS environment is configured correctly. "
+                "See https://boto3.amazonaws.com/v1/documentation/api/latest/guide/quickstart.html#configuration"
+            )
+            raise AmazonBedrockConfigurationError(msg) from exception
+
+        model_adapter_cls = self.get_model_adapter(model=model)
```

Comment:
nit: maybe first check if the given string is a valid AWS model before setting up a connection with AWS?

Reply:
Yeah, that's what `get_model_adapter`does. If not found we'll raise AmazonBedrockConfigurationError